lib/solveEquation/stepThrough.js

```diff
@@ -68,7 +71,42 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
     }
 
     try {
-      equationStatus = step(equation, symbolName);
+      // Find roots of a equation
+      // Conditions: left side is a mul node, right side is 0
+      // TODO: move polynomial to left if is on the right
+      // TODO: assuming a single symbol in equation
+      // TODO: display all solutions
+      // Returns: An equation in the form of symbol = root
+      if (equation.leftNode.op === '*'
+          && (Node.Type.isConstant(equation.rightNode) && parseFloat(equation.rightNode.value) === 0)) {
+        // Gets all factors that aren't a constant
+        const factors = equation.leftNode.args.filter(arg => !Node.Type.isConstant(arg));
+
+        const solutions = [];
+
+        let symbol;
+        // For each factor, solve the equation factor = 0 for the symbol
+        for (var f in factors) {
+          const leftNode = factors[f].content;
```

Comment:
here you're assuming it's a parenthesis node, which it might not be

e.g. (x-2)*x = 0